aesara/link/c/cmodule.py

```diff
@@ -2717,9 +2719,20 @@ def default_blas_ldflags():
 
         # We need to catch warnings as in some cases NumPy print
         # stuff that we don't want the user to see.
-        # I'm not able to remove all printed stuff
-        with warnings.catch_warnings(record=True):
-            numpy.distutils.system_info.system_info.verbosity = 0
+        @contextmanager
+        def catch_numpy_warnings():
+            with warnings.catch_warnings(record=True):
+                numpy.distutils.system_info.system_info.verbosity = 0  # side-effect
```

Comment:
I was wondering this myself. I think that resetting after exiting the context would be cleaner, but for the first step I wanted to preserve the current behavior of the code. (The code currently leaves this set to `0`, so resetting it could lead to other previously suppressed warnings resurfacing.)

Reply:
This is a really good point, but

* fixing this would require changing the existing behavior, which risks breaking lots of other stuff, and
* I don't need to touch this line in order to filter these "Could not locate executable" warnings.

Thus I propose we keep the code as-is for now, and address this in a subsequent PR. So that we don't forget, I opened https://github.com/aesara-devs/aesara/issues/1005.